pkg/cluster/mpi/configmap.go

```diff
@@ -14,12 +15,47 @@ import (
 	"github.com/dominodatalab/distributed-compute-operator/pkg/controller/core"
 )
 
-var (
-	sshdConfigTmpl = `
-Port %d
-AuthorizedKeysFile %s
-`
-)
+const launchScriptTemplate = `#!/bin/bash
+
+set -o nounset 
+set -o errexit
+
+SSH_PORT=%[1]d
+DOMINO_UID=%[2]d
+DOMINO_GID=%[3]d
+DOMINO_USER=%[4]s
+DOMINO_GROUP=%[5]s
+AUTHORIZED_KEYS_PATH=%[6]s
+
+if ! id $DOMINO_UID >/dev/null 2>&1; then
+    groupadd -g $DOMINO_GID $DOMINO_GROUP
+    useradd -u $DOMINO_UID -g $DOMINO_GID -mN -s /bin/bash $DOMINO_USER
+fi
+if [ "$(id -nu $DOMINO_UID)" != "$DOMINO_USER" ]; then
+    echo >&2 "User name mismatch"
```

Comment:
Can we rename the user instead of failing?

Reply:
I've never thought renaming a user is an option in Linux, but looks like `usermod` command can do exactly that... 

Still, I think this is a wrong path: as we're requiring UID and username as CRD arguments (with well-known defaults), a mismatch of those is clearly a violation of the requirement and should cause an exception. Unless of course we find some common scenario of when this happens, in which case we may want to silently rename the user.

Reply:
Hmmm, what requirements are being violated here? The UID and GID we use is arbitrary. If some image happens to already have that user configured, we can probably use it, but we need a specific username to login, which is largely cosmetic.